db/migrate/20200327170640_remove_empty_benchmarks.rb

```diff
@@ -0,0 +1,23 @@
+class RemoveEmptyBenchmarks < ActiveRecord::Migration[5.2]
+  def change
+    ::Xccdf::Benchmark.transaction do
+      empty_benchmarks = ::Xccdf::Benchmark.where.not(
+        id: Profile.select(:benchmark_id).distinct
+      )
+      empty_benchmarks.each do |empty_benchmark|
+        RuleIdentifier.where(
+          rule_id: empty_benchmark.rules.select(:id)
+        ).delete_all
+        empty_rule_references_rules = RuleReferencesRule.where(
+          rule_id: empty_benchmark.rules.select(:id)
+        )
+        RuleReferences.where(
+          id: empty_rule_references_rules.select(:rule_reference_id)
+        ).delete_all
+        empty_rule_references_rules.delete_all
+        empty_benchmark.rules.delete_all
+      end
+      empty_benchmarks.destroy_all
```

Comment:
Line 18 deletes the rules, `         empty_benchmark.rules.delete_all `

I did it like this because otherwise it's making a ton of statements and it is way slower

Reply:
Ok, idk how I missed line 18 :)